### PR TITLE
NyaAccent for cat ears

### PR DESCRIPTION
**Pull request by CaptainSqrBeard**
## About the PR 
Adds NyaAccent for cat ears. I don't like OwOAccent because it makes cat ears considered as something furry, while neko isn't furries.
I also changed "owoify" command description for same reason

NyaAccent adds random words at end of sentences from the list and also replaces some other phrases and words.

UPD: Now it replaces some parts of words, so video is kinda outdated
- "Mo" -> "Meo"
- "Mor" -> "Murr"
- "Per" -> "Purr"

**Screenshots**

https://user-images.githubusercontent.com/56765288/200132082-9dabeca1-1970-4ed5-826c-ea1dd1e6b90c.mp4

**Changelog**

:cl: csqrb
- add: Added Nya Accent
- tweak: Cat ears are now using Nya Accent instead of OwO Accent

### CODE REVIEW

**Comment by 0x6273** [comment]
Just adding cat sounds at the end seems kinda meh. I think it should replace words that sound similar, so "right now" would become "right meow."

**Review by moonheart08 — APPROVED** [review]

**Review by moonheart08 — CHANGES_REQUESTED** [review]
github bait and switch'd me, i thought i was reviewing another PR.

**Comment by CaptainSqrBeard** [comment]
I added more direct replacements and also replacing word parts with other one
- "Mo" -> "Meo"
- "Mor" -> "Murr"
- "Per" -> "Purr"

**Comment by 0x6273** [comment]
image
Please fix your editor so it doesn't do this. We have `insert_final_newline = true` in `.editorconfig`, but your editor/IDE seems to be ignoring that.

File: Content.Server/Speech/EntitySystems/NyaAccentSystem.cs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                // Insert ending word
+                new_s += insert + punct_mark;
+            }
+            final_msg += new_s;
+        }
+
+        // Direct replacements
+        foreach (var (first, replace) in DirectReplacements)
+        {
+            final_msg = Regex.Replace(final_msg, $@"(?<!\w){first}(?!\w)", replace, RegexOptions.IgnoreCase);
```
Comment (moonheart08) [inline Content.Server/Speech/EntitySystems/NyaAccentSystem.cs:90]: Disregarding the fact I dislike the PR contents, you need to think up something smarter than this regex, it's quite expensive and causes the regex engine to repeatedly scan the entire string if, for example, your chat message is a single giant word.

Comment (CaptainSqrBeard) [reply Content.Server/Speech/EntitySystems/NyaAccentSystem.cs:90]: guess nothing bad will happen if i will use simple string.Replace() instead of Regex

Comment (CaptainSqrBeard) [reply Content.Server/Speech/EntitySystems/NyaAccentSystem.cs]: here

**Comment by CaptainSqrBeard** [comment]
I think everything must be fine now. Ready for review

**Comment by mirrorcult** [comment]
i dont see the point of this. you want to wear cat ears without people thinking its a furry thing?

**Comment by mirrorcult** [comment]
to me this is just a worse version of the accent that already exists

**Comment by CaptainSqrBeard** [comment]
They are different. OwO is a furry thing and it's being used by them. Nekos don't use it, they have their own accent. But in SS14 they for some reason share the same accent and i don't like it.

**Comment by mirrorcult** [comment]
we dont have nekos

**Comment by CaptainSqrBeard** [comment]
What do you mean? Cat ears is basically neko cosplay item

**Comment by metalgearsloth** [comment]
Isn't this furry

**Comment by Zumorica** [comment]
Yeah, I think that's pretty furry

**Comment by mirrorcult** [comment]
you will just have to endure being a furry in denial, sorry

**Comment by CaptainSqrBeard** [comment]
💀

**Comment by CaptainSqrBeard** [comment]
Anyway i make it look less furry, but PR is closed anyway.

**Pull request closed without merging**